ush/ioda/bufr2ioda/bufr2ioda_adpsfc_prepbufr.py

```diff
@@ -0,0 +1,341 @@
+    DATA_PATH = os.path.join(dump_dir, f"{cycle_type}.{yyyymmdd}",
+                             str(hh), bufrfile)
+
+    logger.info(f"The DATA_PATH is: {DATA_PATH}")
+
+    # ============================================
+    # Make the QuerySet for all the data we want
+    # ============================================
+    start_time = time.time()
+
+    logger.info('Making QuerySet ...')
+    q = bufr.QuerySet(subsets)
+
+    # ObsType
+    q.add('observationType', '*/TYP')
+
+    # MetaData
+    q.add('stationIdentification', '*/SID')
+    q.add('latitude', '*/YOB')
+    q.add('longitude', '*/XOB')
+    q.add('obsTimeMinusCycleTime', '*/DHR')
+    q.add('heightOfStation', '*/Z___INFO/Z__EVENT{1}/ZOB')
+    q.add('pressure', '*/P___INFO/P__EVENT{1}/POB')
+
+#   # Quality Infomation (Quality Indicator)
+    q.add('qualityMarkerStationPressure', '*/P___INFO/P__EVENT{1}/PQM')
+    q.add('qualityMarkerStationElevation', '*/Z___INFO/Z__EVENT{1}/ZQM')
+
+    # ObsValue
+    q.add('stationPressure', '*/P___INFO/P__EVENT{1}/POB')
+    q.add('stationElevation', '*/ELV')
+
+    end_time = time.time()
+    running_time = end_time - start_time
+    logger.info(f"Running time for making QuerySet: {running_time} seconds")
+
+    # ==============================================================
+    # Open the BUFR file and execute the QuerySet to get ResultSet
+    # Use the ResultSet returned to get numpy arrays of the data
+    # ==============================================================
+    start_time = time.time()
+
+    logger.info(f"Executing QuerySet to get ResultSet ...")
+    with bufr.File(DATA_PATH) as f:
+        r = f.execute(q)
+
+    logger.info(" ... Executing QuerySet: get ObsType ...")
+    # ObsType
+    typ = r.get('observationType')
+
+    logger.info(" ... Executing QuerySet: get MetaData ...")
+    # MetaData
+    sid = r.get('stationIdentification')
+    lat = r.get('latitude')
+    lon = r.get('longitude')
+    lon[lon > 180] -= 360
+    zob = r.get('heightOfStation', type='float')
+    pressure = r.get('pressure')
+    pressure *= 100
+
+    logger.info(f" ... Executing QuerySet: get QualityMarker information ...")
+    # Quality Information
+    pobqm = r.get('qualityMarkerStationPressure')
+    zobqm = r.get('qualityMarkerStationElevation')
+
+    logger.info(f" ... Executing QuerySet: get obsvalue: stationPressure ...")
+    # ObsValue
+    elv = r.get('stationElevation', type='float')
+    pob = r.get('stationPressure')
+    pob *= 100
+
+    logger.info(f" ... Executing QuerySet: get datatime: observation time ...")
+    # DateTime: seconds since Epoch time
+    # IODA has no support for numpy datetime arrays dtype=datetime64[s]
+    dhr = r.get('obsTimeMinusCycleTime', type='int64')
+
+    logger.info(f" ... Executing QuerySet: Done!")
+
+    logger.info(f" ... Executing QuerySet: Check BUFR variable generic \
+                dimension and type ...")
+    # Check BUFR variable generic dimension and type
```

Comment:
Do we need all of this printed for `info`? Can we make it `debug`?

Reply:
Yes, we can make it debug. I'll change that.